servers/src/grin/sync.rs

```diff
@@ -141,18 +141,34 @@ pub fn run_sync(
 
 						// run the header sync every 10s
 						if si.header_sync_due(&header_head) {
-							header_sync(peers.clone(), chain.clone(), &mut history_locators);
-
 							let status = sync_state.status();
-							match status {
-								SyncStatus::TxHashsetDownload => (),
-								_ => {
-									sync_state.update(SyncStatus::HeaderSync {
-										current_height: header_head.height,
-										highest_height: si.highest_height,
-									});
+							let update_sync_state = match status {
```

Comment:
Perhaps move this section to its own function to keep the loop relatively clean?